rental_handler.rb

```diff
@@ -0,0 +1,38 @@
+require_relative 'rental'
+
+class RentalHandler
+  def initialize(persons, books)
+    @rentals = []
+    @books = books.books
+    @persons = persons.persons
+  end
+
+  def rent_book
+    puts 'Select a book from the following list by number'
+    @books.each_with_index do |book, index|
+      puts "#{index}. Title: #{book.title} Author: #{book.author}"
+    end
+    book_index = gets.chomp.to_i
+    puts 'Select a person from the following list by number'
+    @persons.each_with_index do |person, index|
+      puts "#{index}. Name: #{person.name} Age: #{person.age}"
+    end
+    person_index = gets.chomp.to_i
+    puts 'Date: '
+    date = gets.chomp
+    @rentals << Rental.new(date, @books[book_index], @persons[person_index])
+    puts 'Book rented successfully'
+  end
```

Comment:
- [ ] @Team Great work adding a method to add a new rental, however, to achieve the Single-responsibility Principle all requests for input and input validation should've been placed in the main.rb file and/or classes/methods that are responsible for requesting or validating input. You can add a new file whose main role is to ensure the user can add new persons.

Reply:
solve